qiskit/transpiler/passes/optimization/template_matching/forward_match.py

```diff
@@ -404,13 +404,31 @@ def run_forward_match(self):
                 ):
                     continue
 
+                # Check if parameters match the template or not.
+                #Construct a temporary list of matches. 
+                temp_match = self.match.copy()
+
+                # Append the new match to the list of matches.
+                temp_match.append([i,label])
+
+                substitution = TemplateSubstitution(
+                    [temp_match],
+                    self.circuit_dag_dep,
+                    self.template_dag_dep,
+                    None
+                )
+                    
+                template_sublist = [x[0] for x in temp_match]
+                circuit_sublist = [x[1] for x in temp_match]
+                template = substitution._attempt_bind(template_sublist,circuit_sublist)
+
                 # Check if the qubit, clbit configuration are compatible for a match,
                 # also check if the operation are the same.
                 if (
                     self._is_same_q_conf(node_circuit, node_template)
                     and self._is_same_c_conf(node_circuit, node_template)
                     and self._is_same_op(node_circuit, node_template)
-                ):
+                ) and template is not None :
```

Comment:
Instead of adding a condition to this if statement, add a block before it like this:
```python
if template is None:
    continue
```
This is more consistent with the existing code.

Reply:
I fixed at the next version.